elements/pl-card/pl-card.py

```diff
@@ -0,0 +1,71 @@
+import chevron
+import lxml.html
+import prairielearn as pl
+
+HEADER_DEFAULT = ""
+TITLE_DEFAULT = ""
+SUBTITLE_DEFAULT = ""
+FOOTER_DEFAULT = ""
+IMG_TOP_SRC_DEFAULT = ""
+IMG_TOP_ALT_DEFAULT = ""
+IMG_BOTTOM_SRC_DEFAULT = ""
+IMG_BOTTOM_ALT_DEFAULT = ""
+WIDTH_DEFAULT = "auto"
+
+
+def prepare(element_html: str, data: pl.QuestionData) -> None:
+    element = lxml.html.fragment_fromstring(element_html)
+    required_attribs = []
+    optional_attribs = [
+        "header",
+        "title",
+        "subtitle",
+        "footer",
+        "img-top-src",
+        "img-top-alt",
+        "img-bottom-src",
+        "img-bottom-alt",
+        "width",
+    ]
+    pl.check_attribs(element, required_attribs, optional_attribs)
+
+
+def render(element_html: str, data: pl.QuestionData) -> str:
+    element = lxml.html.fragment_fromstring(element_html)
+
+    header = pl.get_string_attrib(element, "header", HEADER_DEFAULT)
+    title = pl.get_string_attrib(element, "title", TITLE_DEFAULT)
+    subtitle = pl.get_string_attrib(element, "subtitle", SUBTITLE_DEFAULT)
+    footer = pl.get_string_attrib(element, "footer", FOOTER_DEFAULT)
+    img_top_src = pl.get_string_attrib(element, "img-top-src", IMG_TOP_SRC_DEFAULT)
+    img_top_alt = pl.get_string_attrib(element, "img-top-alt", IMG_TOP_ALT_DEFAULT)
+    img_bottom_src = pl.get_string_attrib(
+        element, "img-bottom-src", IMG_BOTTOM_SRC_DEFAULT
+    )
+    img_bottom_alt = pl.get_string_attrib(
+        element, "img-bottom-alt", IMG_BOTTOM_ALT_DEFAULT
+    )
+    width = pl.get_string_attrib(element, "width", WIDTH_DEFAULT)
```

Comment:
Width can't have an arbitrary value (e.g. I can't put `53%`), so I'd rather make this an enum attribute to alert instructors to invalid values. @eliotwrobson all other enum attributes I've seen so far are using string values without special characters; is it possible to have an enum for `25%`/`50%`/etc.? If not, we can still do manual validation with a check like `width in {"25%", "50%", "75%", "auto"}`.

Reply:
Yeah, the enum getter helper function doesn't allow a percentage sign in the name or starting with a number, so "50%" couldn't be used as the name. Although I'd argue that spelling out "half" is more intuitive, since having a percentage sign in the string makes it seem like "53%" would be allowed.

Reply:
Gotcha! I'm wary of spelling it out - `one-quarter` and `three-quarters` seems weird to me. But either way, we should still validate the attribute value.

Reply:
@nwalters512 https://github.com/PrairieLearn/PrairieLearn/pull/6978/commits/83b549bb1bf38b0f63bc2b50bd87a5099a8ea7a0